jetty-annotations/src/main/java/org/eclipse/jetty/annotations/AnnotationParser.java

```diff
@@ -68,10 +68,30 @@
 {
     private static final Logger LOG = Log.getLogger(AnnotationParser.class);
 
+    private static final int JVM_MAJOR_VER;
+
     protected Set<String> _parsedClassNames = ConcurrentHashMap.newKeySet();
     
     protected static int ASM_OPCODE_VERSION = Opcodes.ASM5; //compatibility of api
-   
+
+    static
+    {
+        // Determine JVM spec version
+        // Using guidance from http://openjdk.java.net/jeps/223
+        String jvmSpecVer = System.getProperty("java.vm.specification.version");
+
+        if (jvmSpecVer.indexOf('.') >= 0)
```

Comment:
Note that as part of my work on Conscrypt, I am moving a copy of JavaVersion from jetty-start to jetty-util (and adding a static final instance for the current JVM).  See the jetty-9.4.x-conscrypt-alpn branch.

We should do this instead of letting bespoke version parsing propagate throughout the code.   Feel free to do this separately rather than wait for conscrypt.

Reply:
Sure, sounds good.  We can make the change to use the jetty-util version in a later commit.